chains/evm/calls/erc721/erc721.go

```diff
@@ -0,0 +1,204 @@
+package erc721
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/consts"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/transactor"
+	"github.com/ChainSafe/chainbridge-core/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog/log"
+)
+
+type ERC721Contract struct {
+	client                calls.ContractCallerDispatcherClient
+	erc721ContractAddress common.Address
+	abi                   abi.ABI
+	transactor.Transactor
+}
+
+func NewErc721Contract(
+	client calls.ContractCallerDispatcherClient,
+	erc721ContractAddress common.Address,
+	t transactor.Transactor,
+) *ERC721Contract {
+	// load ABI
+	a, err := abi.JSON(strings.NewReader(consts.ERC721PresetMinterPauserABI))
+	if err != nil {
+		log.Fatal().Msg("Unable to load ABI") // TODO
+	}
+	return &ERC721Contract{
+		client:                client,
+		erc721ContractAddress: erc721ContractAddress,
+		abi:                   a,
+		Transactor:            t,
+	}
+}
+
+func (c *ERC721Contract) AddMinter(minter common.Address, opts transactor.TransactOptions) (*common.Hash, error) {
+	addMinterInput, err := c.prepareErc721AddMinterInput(minter)
+	if err != nil {
+		log.Error().Err(err)
+		return nil, err
+	}
+
+	txHash, err := c.Transact(&c.erc721ContractAddress, addMinterInput, opts)
+
+	if err != nil {
+		log.Error().Err(err)
+		return nil, err
+	}
+	return txHash, err
+}
+
+func (c *ERC721Contract) Approve(tokenId *big.Int, recipient common.Address, opts transactor.TransactOptions) (*common.Hash, error) {
+	approveTokenInput, err := c.prepareERC721ApproveInput(recipient, tokenId)
+	if err != nil {
+		log.Error().Err(fmt.Errorf("erc721 approve input error: %v", err))
+		return nil, err
+	}
+
+	txHash, err := c.Transact(&c.erc721ContractAddress, approveTokenInput, opts)
+	if err != nil {
+		log.Error().Err(err)
+		return nil, err
+	}
+	return txHash, err
+}
+
+func (c *ERC721Contract) Deposit(tokenId *big.Int, metadata string, destinationId int, resourceId types.ResourceID, bridgeContract, recipient common.Address, opts transactor.TransactOptions) (*common.Hash, error) {
```

Comment:
This method doesn't make sense here - it will be moved to bridge contract as I refactor everything